### ISSUE TITLE
error: no matching member function for call to insert

### ISSUE THREAD

**Issue by xiota** [minecraft-linux/mcpelauncher-manifest#1028]
**Describe the bug**
Fails to build.  Using `branch-sync-mcpelauncher-qt6`.  Probably missing some header?
```
src/mcpelauncher-manifest/libc-shim/src/cstdio.cpp:140:10: error: no matching member function for call to 'insert'
```

Clang 18.1.8:
```
-- The C compiler identification is Clang 18.1.8
-- The CXX compiler identification is Clang 18.1.8
```

**To Reproduce**
Clone, checkout branch, git submodules, configure/build with cmake, etc.

**Expected behavior**
Build should succeed.

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Version: 09aa55e68acde708519151a013cde361eea1f74a
 - Installation Type: N/A

**Comment by ChristopherHX** [comment]
>  Using GCC 14.1.1

That compiler is probably not used, gcc support has been dropped in 2020 and never restored.
AUR and all other builds use clang.

This could be indeed a libstdc++ change, but usually missing includes caused better error messages that could be resolved without me installing bleeding edge toolchains to a ubuntu lts system where this compiles like before

**Comment by xiota** [comment]
Sorry for the mistake.  Editing original description to correct.  Compiler is clang 18.
```
-- The C compiler identification is Clang 18.1.8
-- The CXX compiler identification is Clang 18.1.8
```

Another package had similar error with different function, fixed by adding missing header, on GCC 14.  So I assumed it was the same compiler.  Apparently, GCC 14 and Clang 18 are both more strict about headers with less helpful error messages.

**Comment by jplatte** [comment]
Since this happens when compiling the `libc-shim` library, I opened an issue (with some extra detail) over there: https://github.com/minecraft-linux/libc-shim/issues/36

**Comment by xiota** [comment]
Problem is caused by `-Wp,-D_FORTIFY_SOURCE=3` in the default build flags on Arch Linux.  Removing them fixes the problem, allowing build to complete successfully.  Possibly changing the value to 1 or 2 would also work, but I did not try.

Closing because the existence of this issue with solution for users to find is sufficient to resolve the problem.

Related: https://github.com/minecraft-linux/libc-shim/issues/33
